pyteal/ast/scratchvar.py

```diff
@@ -17,7 +17,15 @@ class ScratchVar:
             ])
     """
 
-    def __init__(self, type: TealType = TealType.anytype, slotId: int = None):
+    MAX_STACK_DEPTH = 700 * 256 / 2
+
+    def __init__(
+        self,
+        type: TealType = TealType.anytype,
+        slotId: int = None,
+        *,
+        slodIdFromStack: bool = False,
```

Comment:
```suggestion
        slotId: Union[int, Expr] = None,
```
Based on the conversation we had, I suggest doing something like this so that you'll be able to pass in the slot to use with the dynamic version of the opcodes. I imagine this could work like this:
* If `slotId` is an int, then `store` & `load` should be used to access it like they do currently
* If `slotId` is an `Expr`, then is represents a dynamic, runtime index, so `stores` and `loads` should be used to access it.

Reply:
Also I'd probably recommend making the static vs dynamic decision inside of `ScratchVar` so that `ScratchSlot` does not change too much since it's used extensively in the compiler as a way to represent a static slot. Maybe a `DynamicScratchSlot` is needed to address the dynamic case?